### PR TITLE
Python: Project setup

### PR DESCRIPTION
**Pull request by lucasmcdonald3**
_Issue #, if available:_

_Description of changes:_

_Squash/merge commit message, if applicable:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Initialize generated Dafny
+from .internaldafny.generated import module_
+
+# Initialize externs
+from .internaldafny import extern
+
+# Export user-friendly access paths
+import aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders as mpl
+import aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders as keystore
```
Comment (justplaz) [inline AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/__init__.py:12]: this seems like a typo, 
i'm guessing it should be the keystore and not the materialproviders module here? 

`import aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_keystore as keystore`

File: ComAmazonawsKms/Makefile [hunk 2]
```diff
@@ -13,3 +13,13 @@ MAX_RESOURCE_COUNT=10000000
 AWS_SDK_CMD := --aws-sdk
 STD_LIBRARY=StandardLibrary
 SMITHY_DEPS=model
+
+format_net:
+	pushd runtimes/net && dotnet format && popd
```
Comment (justplaz) [inline ComAmazonawsKms/Makefile:18]: question: do we need/want to do this for ddb as well?

File: TestVectorsAwsCryptographicMaterialProviders/runtimes/python/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+version = "0.1.0"
+description = ""
+authors = ["AWS Crypto Tools <aws-crypto-tools@amazon.com>"]
+packages = [
+    { include = "aws_cryptography_materialproviderstestvectorkeys", from = "src" }
+]
+
+[tool.poetry.dependencies]
+python = "^3.11.0"
+# TODO-Python: Not sure why this is needed... I think boto3 uses 0.19.14, but it's not resolving on PyPI atm
```
Comment (justplaz) [inline TestVectorsAwsCryptographicMaterialProviders/runtimes/python/pyproject.toml:12]: is this TODO still accurate?

File: ComAmazonawsKms/Makefile (hunk 2 above)
Comment (lucasmcdonald3) [reply ComAmazonawsKms/Makefile:18]: Oh, I didn't mean to add this... this was probably a merge issue.
I'll resolve this to whatever's on main.

File: TestVectorsAwsCryptographicMaterialProviders/runtimes/python/pyproject.toml (hunk 3 above)
Comment (lucasmcdonald3) [reply TestVectorsAwsCryptographicMaterialProviders/runtimes/python/pyproject.toml:12]: Good catch. Will look into this.

File: AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/__init__.py (hunk 1 above)
Comment (lucasmcdonald3) [reply AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/__init__.py:12]: Wow, yeah. Will fix.

File: TestVectorsAwsCryptographicMaterialProviders/runtimes/python/pyproject.toml (hunk 3 above)
Comment (lucasmcdonald3) [reply TestVectorsAwsCryptographicMaterialProviders/runtimes/python/pyproject.toml:12]: Seems not needed, will remove.

File: ComAmazonawsKms/Makefile (hunk 2 above)
Comment (lucasmcdonald3) [reply ComAmazonawsKms/Makefile:18]: Removed, I believe this was moved into the general SmithyDafnyMakefile.

File: AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/__init__.py (hunk 1 above)
Comment (lucasmcdonald3) [reply AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/__init__.py]: Fixed

**Review by justplaz — APPROVED** [review]
LGTM

**Pull request merged**